airflow-core/src/airflow/models/dag.py

```diff
@@ -136,6 +136,8 @@ def get_run_data_interval(timetable: Timetable, run: DagRun) -> DataInterval:
         return data_interval
     # Compatibility: runs created before AIP-39 implementation don't have an
     # explicit data interval. Try to infer from the logical date.
+    if TYPE_CHECKING:
+        assert run.logical_date
```

Comment:
```suggestion
        assert run.logical_date is None
```

it might a bit be easier to read this way

Reply:
This should be `is not None`

Reply:
right 🤦‍♂️